### PR TITLE
[SPARK-47945][SQL] MsSQLServer: Document Mapping Spark SQL Data Types from Microsoft SQL Server and add tests

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR added  Document Mapping Spark SQL Data Types from Microsoft SQL Server and added more tests
### Why are the changes needed?

doc improvement

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

new test and doc build
image

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Could you resolve the conflicts?

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala [hunk 1]
```diff
@@ -265,7 +273,7 @@ class MsSqlServerIntegrationSuite extends DockerJDBCIntegrationSuite {
     assert(types(7).equals("class java.lang.String"))
     assert(types(8).equals("class [B"))
     assert(row.getString(0).length == 10)
-    assert(row.getString(0).trim.equals("the"))
+    assert(row.getString(0).equals("the".padTo(10, ' ')))
```
Comment (dongjoon-hyun) [inline connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala]: This seems to be ANSI result. Is this safe in non-ANSI CI, @yaooqinn ?
- https://github.com/apache/spark/actions/workflows/build_non_ansi.yml

File: docs/sql-data-sources-jdbc.md [hunk 2]
```diff
@@ -1441,3 +1441,192 @@ The Spark Catalyst data types below are not supported with suitable Oracle types
...
+      <td>(Default)preferTimestampNTZ=false or spark.sql.timestampType=TIMESTAMP_LTZ</td>
+    </tr>
+    <tr>
+      <td>datetime2 [ (fractional seconds precision) ]</td>
+      <td>TimestampNTZType</td>
+      <td>preferTimestampNTZ=true or spark.sql.timestampType=TIMESTAMP_NTZ</td>
+    </tr>
+    <tr>
+      <td>datetimeoffset [ (fractional seconds precision) ]</td>
+      <td>StringType</td>
```
Comment (yaooqinn) [inline docs/sql-data-sources-jdbc.md:1543]: https://github.com/apache/spark/blob/9d715ba491710969340d9e8a49a21d11f51ef7d3/sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala#L112-L114

This comment appears not true as we use mssql-jdbc

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala (hunk 1 above)
Comment (yaooqinn) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala]: Char padding is ANSI-irrelevant, AFAIK

Comment (yaooqinn) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala]: Anyway, I removed this LOC

Comment (dongjoon-hyun) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala]: Thanks. Yes, that would be safe.

File: docs/sql-data-sources-jdbc.md (hunk 2 above)
Comment (dongjoon-hyun) [reply docs/sql-data-sources-jdbc.md:1543]: Ya, let's take a look at that separately.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM for the AS-IS PR. Thank you, @yaooqinn .

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun 

Merged to master

**Pull request closed without merging**